src/modules/file/file.js

```diff
@@ -94,3 +94,277 @@ function file_entity_field_formatter_view(entity_type, entity, field, instance,
   }
   catch (error) { console.log('file_entity_field_formatter_view - ' + error); }
 }
+
+/**
+ * Implements hook_field_widget_form().
+ */
+function file_field_widget_form(form, form_state, field, instance, langcode, items, delta, element) {
+  try {
+    // Change the item type to a hidden input to hold the file id.
+    // items[delta].type = 'textfield';
+    items[delta].type = 'hidden';
+
+    // If we already have an file for this item, show it.
+    // @TODO: show exisiting fid
+
+    var browse_button_text = t('Add Media');
+    var item_id_base = items[delta].id.replace(/-/g, '_');
+
+    var browse_button_id = items[delta].id + '-add-media-button';
+
+    var html = '<div id="' + items[delta].id + '-media"></div><a href="#" data-role="button" data-icon="camera" ' +
+      'id="' + browse_button_id + '" ' +
+      'data-form_id="' + form.id + '" ' +
+      'data-name="' + element.name + '" ' +
+      'data-delta="' + delta + '" ' +
+      'data-cardinality="' + field.cardinality + '" ' +
+      '>' +
+      browse_button_text +
+      '</a>';
+
+
+
+
+    // Open extra javascript declaration.
+    //
+// $("#" + drupalgap_get_page_id(drupalgap_path_get())).on("pageshow", function () {
+//   document.addEventListener("deviceready", init, false);
+// });
+
+
+    html += '<script type="text/javascript">';
+    // Add device ready listener for PhoneGap camera.
+    html += '$("#' + drupalgap_get_page_id(
```

Comment:
@luxio The jQuery selector here can probably just be `'#' + drupalgap_get_page_id()`